kits/daisy/src/robot/hexapod.cpp

```diff
@@ -533,25 +535,26 @@ Hexapod::Hexapod(std::shared_ptr<Group> group,
         leg_count++;
         // Get from base of each leg
         int num_prev_legs = std::count_if(real_legs_.begin(), real_legs_.end(), [i](int other_leg) { return other_leg < i; });
-        auto accel = fbk[num_prev_legs * num_leg_joints].imu().accelerometer().get(); 
-        auto gyro = fbk[num_prev_legs * num_leg_joints].imu().gyro().get(); 
-        Eigen::Vector3d mod_accel;
-        mod_accel << accel.getX(), accel.getY(), accel.getZ();
-        Eigen::Vector3d mod_gyro;
-        mod_gyro << gyro.getX(), gyro.getY(), gyro.getZ();
+
+        // HEBI Quaternion
+        auto mod_orientation = fbk[num_prev_legs * num_leg_joints]
+          .imu().orientation().get();
+        // Eigen Quaternion
+        Eigen::Quaterniond mod_orientation_eig(
+          mod_orientation.getW(),
+          mod_orientation.getX(),
+          mod_orientation.getY(),
+          mod_orientation.getZ());
+        Eigen::Matrix3d mod_orientation_mat = mod_orientation_eig.toRotationMatrix();
+
         // Transform
         Eigen::Matrix4d trans = legs_[i]->getKinematics().getBaseFrame();
-        avg_accel += trans.topLeftCorner<3,3>() * mod_accel;
-        avg_gyro += trans.topLeftCorner<3,3>() * mod_gyro;
+        avg_grav += trans.topLeftCorner<3,3>() * mod_orientation_mat.transpose() * down;
       }
-      // TODO: remove the 2.0!  This just seems to stabilize the motion right now, but probably because
-      // either:
-      // 1) there are other factors we don't have in there yet (e.g., grav comp)
-      // 2) the jacobian is slightly noisy
-      // Note that the foot forces have been compared to MATLAB results, and seem
-      // reasonable. No further debugging has been done.
-      avg_accel /= num_legs_used * 2.0;
-      avg_gyro /= num_legs_used * 2.0;
+      avg_grav /= num_legs_used;
+      // TODO: ideally, put a lock around this...but this is the only place
+      // we write to it, so it isn't terrible.
+      gravity_direction_ = avg_grav;
```

Comment:
**Explicitly comment at declaration of this variable that it is not thread safe**